File: src/ListSpace.js
'use strict';

function ListSpace(name='', cost=0, address='', description=''){
  this._name = name;
  this._cost = cost;
  this._address = address;
  this._description = description;
};

ListSpace.prototype.all = function(){
  var pg = require(‘pg’);
  var connectionString = "postgres://Student:@localhost/ip:5432/makersbnb_test";
  var pgClient = new pg.Client(connectionString);
  var query = pgClient.query('SELECT * FROM listings;');
  query
};

File: src/ListSpace.rb
require_relative 'database_connection'

class ListSpace
  attr_reader :name, :cost, :address, :description

   def initialize(name:, cost:, address:, description:)
     @name, @cost, @address, @description  = name, cost, address, description
   end

  def self.all
    result = DatabaseConnection.query('SELECT * FROM listings;')
    result.map do |listing|
      ListSpace.new(name: listing['name'], cost: listing['cost'], address: listing['address'], description: listing['description'])
    end
  end

  def self.create(name:, cost:, address:, description:)
    return false unless is_url?(url)
    result = DatabaseConnection.query("INSERT INTO listings (name, cost, address, description) VALUES('#{name}', '#{cost}', '#{address}', '#{description}') RETURNING name, cost, address, description;")
    ListSpace.new(name: listing['name'], cost: listing['cost'], address: listing['address'], description: listing['description'])
  end

end
